File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpContainerPutBatchRequest.java
package calebxzhou.rdi.mc.common2.mcp;

import java.util.List;

public record RMcpContainerPutBatchRequest(
        List<Move> moves,
        boolean dryRun,
        boolean stopOnError
) {
    public record Move(
            Integer fromInventorySlot,
            RMcpContainerMoveBatchRequest.Endpoint to,
            int count
    ) {
    }
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpContainerPutBatchRequest.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RMcpContainerPutBatchRequest(
    val moves: List<Move>,
    val dryRun: Boolean,
    val stopOnError: Boolean
) {
    @JvmRecord
    data class Move(
        val fromInventorySlot: Int,
        val to: RMcpContainerMoveBatchRequest.Endpoint,
        val count: Int
    )
}
